timJohnson/Homework_Week2.js

```diff
@@ -0,0 +1,232 @@
+					resultString += (letterArray[j]);
+				}
+			}		
+		return resultString;
+		}
+    var triangle = letterTriangle(10);
+	console.log(triangle);
+	
+3b)	// Returns all formatted lyrics to 'The Twelve Days of Christmas'
+	function entireSong(){
+	var title = 'Twelve Days of Christmas' + '\n';
+	var songString = '';
+	for (var i=0; i <= 11; i++){
+		songString += '\n' + start(i) + theTwelve(i);
+		}
+	return title + songString;
+	}
+
+	// start function returns the day the song is currently at with the starting lyrics for each verse
+	function start(i){
+	var day = ['first', 'second', 'third', 'fourth', 'fifth', 'sixth', 'seventh', 'eighth', 'ninth', 'tenth', 'eleventh', 'twelfth'];
+	var startLyrics = ['On the' + ' ' + day[i] + ' day of Christmas' + ' \n' + 'my true love sent to me:' + '\n'];
+	return startLyrics;
+	}
+	
+	// theTwelve function returns all lyrics for each day
+	function theTwelve(i){
+		var theTwelve = ['and a Partridge in a Pear Tree', '2 Turtle Doves',  '3 French Hens', '4 Calling Birds', '5 Golden Rings', '6 Geese a Laying', '7 Swans a Swimming', '8 Maids a Milking', '9 Ladies Dancing', '10 Lords a Leaping', '11 Pipers Piping',];      	
+		var theTwelveString = '';
+    if (i === 0){ 
+      theTwelve[0] = 'A Partridge in a Pear Tree';
+    }
+    for (var j=i; j>=0; j--)
+			theTwelveString += theTwelve[j] + '\n';
+		return theTwelveString;
+	}
+	// set song = the function return value
+	var song = entireSong();
+	// display results
+	console.log(song)
+	
+4a)	// Returns true if a and b are true and false if either a or b are false
+	function and2(a,b){
+		return a ? (b ? true : false) : false;
+	}
+	and2(1,4);
+	and2('x'/2, 2);	
+	}
+
+		
+4b)	
+	function and2(a,b){
+	return a ? (b ? true : false) : false;
+	}
+	// Returns true if a, b, and c are true and false if any of a, b, b are false by calling and2 function to get truth about a and b then compare against c
+	function and3(a,b,c){
+		return and2(a,b)? (c ? true : false) : false;
+	}
+	and3(1,3,4); // true
+	and3(1, 2<1, 4); // false
+	and3(0<-1, 1,4); // false
+	and3(1, 2, 4<3); // false
+	
+4c)
+	// Function compareArray returns truth comparisons for any number of inputs via an array.
+	var compareArray = [1,2,4,5,6];
+	function andN(compareArray){
+		if (compareArray.length === 0)
+			return 0;
+		else if (compareArray.length ===1)
+			return compareArray[0] == true;
+		else{
+			var truth = true;
+				for (i=0; i <= (compareArray.length -1); i++){
+						truth = truth ? (compareArray[i] ? true : false) : false;
+				}
+			}
+		return truth;
+	}
+	var truth = andN(compareArray);
+	console.log(truth);
+	
+4d)
+	The function in 4c does not represent all comparisons of &&.  Some of these return values such as NaN, undefined, and empty string
+	e.g.
+	NaN && true;  //returns NaN
+	undefied && 0; //returns undefined
+	'string' && ' '; //returns " "
+	
+5a)	// setCards function creates an array of cards like Ace Of Hearts, Ace of Diamonds... 
```

Comment:
You've got a few problems happening in this code, but I think it's moving along its path correctly. For example, if you try rank(0), your function is currently returning 0, instead of 1. I would suggest running these functions against the suite of test assertions Dan has created in the next problem, and trying to track down where the errors are stemming from. Remember to give some time to work on 5b as well. Testing and checking for validity of your arguments is extremely important!